### PR TITLE
fix: list-task with multiline desc

### PR DESCRIPTION
**Pull request by vmaerten**
Linked to : 
- https://github.com/go-task/task/issues/1637

I have multiple solutions in mind
- First solution : remove all \n so we display one line
image

- Second solution : Display only the first line. We could considere for longer desc, we should use summary and not desc
image

- Third solution: Display "as it" with multiline
image

I would considere that desc should be short and summary for long description. So, I would go for the first one.

### CODE REVIEW

**Comment by pd93** [comment]
Also related:

- #1350

I generally agree that descriptions should be short and kept to a single line. However, I suspect we will have users complain about options 1 and 2 as it restricts their options. A lot of people use the `--list` functionality as their docs and I can understand that restricting this to 1 line is annoying.

I've had a ton of issues using `text/tabwriter` in the past and gave up on trying to make it work. (for example `task --experiements` is still not aligned correctly). If you're able to find a good generalised solution for alignment, I'd be very interested.

**Comment by vmaerten** [comment]
@pd93 I agree, I was not aware of this issue.

I've changed to the solution 3. I've also fixed the issue with experiments display. Now, all experiments are aligned in a table.
image

It seems tabwrite does not work well with colors (cf : https://stackoverflow.com/questions/35398497/how-do-i-get-colors-to-work-with-golang-tabwriter / https://github.com/fatih/color/issues/175 / https://github.com/golang/go/issues/12073)

That why, it can be trick, as we display multiple colors in the same table.

It may be worthy to take a look at https://github.com/juju/ansiterm/
I'll take a look if I can find something that can handle both tabs and colors

**Comment by vmaerten** [comment]
I've push a last version, using https://github.com/Ladicle/tabwriter as a drop in replacement of tabwriter but supporting color

It's way easier to use (no more headache debugging the tabwrite, yes I have a headache now :joy: ) than the text/tabwriter

File: testdata/desc/Taskfile.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+version: 3
+
+
```
Comment (andreynering) [inline testdata/desc/Taskfile.yml:3]: Let's remove these duplicated empty lines.

```suggestion

```

File: testdata/desc/Taskfile.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+tasks:
+  build:
+    aliases:
+      - b
+    desc: |
+      Multi-line escription with alias which is super long long long long long long
+      another line
+      third line long long long long long long long long
+
+
```
Comment (andreynering) [inline testdata/desc/Taskfile.yml:13]:
```suggestion

```

**Review by andreynering — COMMENTED** [review]
I think I prefer option 1, forcing descriptions to be a single line. We have `summary` for longer documentation.

Thoughts?

**Comment by vmaerten** [comment]
I totally agree with you @andreynering. I did this at the first time before the @pd93's comment
We could write in the doc that desc will be printed in one line. If you want multiple lines you need to use summary

**Comment by pd93** [comment]
I'm happy to go with option 1 if you both agree. I do think we will get some complains about this though. I generally prefer to encourage users to doing something a specific way

**Comment by vmaerten** [comment]
Option 1 has been chosen.
Here is the result:

image

**Review by pd93 — APPROVED** [review]

**Pull request merged**